sdk/storage/Azure.Storage.Common/src/Shared/StorageResponseClassifier.cs

```diff
@@ -23,10 +22,24 @@ public StorageResponseClassifier(Uri secondaryStorageUri)
         /// <returns></returns>
         public override bool IsRetriableResponse(HttpMessage message)
         {
-            if (message.Request.Uri.Host == SecondaryStorageUri.Host && message.Response.Status == Constants.HttpStatusCode.NotFound)
+            if (message.Request.Uri.Host == SecondaryStorageUri?.Host && message.Response.Status == Constants.HttpStatusCode.NotFound)
```

Comment:
Add a comment here and reword the comment on the method now that this is doing more.

Reply:
Should we just remove the ctor since SecondaryStorageUri is no longer required?

Reply:
That's probably a good idea.

Reply:
Updated